internal/common/client_test.go

```diff
@@ -0,0 +1,44 @@
+package common
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"gonum.org/v1/gonum/mat"
+	"testing"
+)
+
+// TestBasicRuleEvaluatorPositive Checks whether rule we expect to evaluate as true actually evaluates as such
+func TestBasicRuleEvaluatorPositive(t *testing.T) {
+	result, err := BasicRuleEvaluator("Kinda Complicated Rule")
+	if !result {
+		t.Errorf("Rule evaluation came as false, when it waqs expected to be true, potential error with value '%v'", err)
+	}
+}
+
+// TestBasicRuleEvaluatorNegative Checks whether rule we expect to evaluate as false actually evaluates as such
+func TestBasicRuleEvaluatorNegative(t *testing.T) {
+	registerTestRule()
+	result, err := BasicRuleEvaluator("Kinda Test Rule")
+	if result || err != nil {
+		t.Errorf("Rule evaluation came as true, when it was expected to be false, potential error with value '%v'", err)
+	}
+}
+
+func registerTestRule() {
+
+	//A very contrived rule//
+	name := "Kinda Test Rule"
+	reqVar := []string{
+		"number_of_islands_contributing_to_common_pool",
```

Comment:
Perhaps we should change the "magic strings" defining the rule components with an enum?

Reply:
as per https://github.com/SOMAS2020/SOMAS2020/pull/14/files#r533407903 :)

In convo with @Neelesh99  i think eventually these will be constants in a file